### PR TITLE
New: Improved Error Messages for Indexer Test/Add Failures

### LINKED ISSUE #648: Improve Cardigann Error Message when adding/editing an indexer
We already have a cardigann exception, we should throw this to the GUI.

Same for all other indexers.

image

```
2021-11-28 13:58:08.7|Warn|Cardigann|Unable to connect to indexer

[v0.1.5.1157] NzbDrone.Core.Indexers.Definitions.Cardigann.Exceptions.CardigannConfigException: Error: Invalid Username/password combination!
   at NzbDrone.Core.Indexers.Cardigann.CardigannRequestGenerator.CheckForError(HttpResponse loginResult, IList`1 errorBlocks) in D:\a\1\s\src\NzbDrone.Core\Indexers\Definitions\Cardigann\CardigannRequestGenerator.cs:line 540
   at NzbDrone.Core.Indexers.Cardigann.CardigannRequestGenerator.DoLogin() in D:\a\1\s\src\NzbDrone.Core\Indexers\Definitions\Cardigann\CardigannRequestGenerator.cs:line 204
   at NzbDrone.Core.Indexers.Cardigann.Cardigann.DoLogin() in D:\a\1\s\src\NzbDrone.Core\Indexers\Definitions\Cardigann\Cardigann.cs:line 159
   at NzbDrone.Core.Indexers.HttpIndexerBase`1.FetchIndexerResponse(IndexerRequest request) in D:\a\1\s\src\NzbDrone.Core\Indexers\HttpIndexerBase.cs:line 388
   at NzbDrone.Core.Indexers.HttpIndexerBase`1.FetchPage(IndexerRequest request, IParseIndexerResponse parser) in D:\a\1\s\src\NzbDrone.Core\Indexers\HttpIndexerBase.cs:line 320
   at NzbDrone.Core.Indexers.HttpIndexerBase`1.TestConnection() in D:\a\1\s\src\NzbDrone.Core\Indexers\HttpIndexerBase.cs:line 456


2021-11-28 13:58:08.7|Error|Microsoft.AspNetCore.Diagnostics.ExceptionHandlerMiddleware|An unhandled exception has occurred while executing the request.

[v0.1.5.1157] FluentValidation.ValidationException: Validation failed: 
 -- : Unable to connect to indexer, check the log for more details
```

### PR DESCRIPTION
**Pull request by bakerboy448**
removed unnecessary usings
Fixes #648

#### Database Migration
NO

#### Description
- Switches from generic `check the logs` error messages to catching common exceptions
- indexer test catches http & web exception errors
- Cardigann exceptions base changed to indexer exceptions

Generally improves errors to be clear to the user rather than directing them to logs which may be confusing and contain lots of spew

#### Todos
- Tests
- Translation Keys (./src/NzbDrone.Core/Localization/Core/en.json)
- [Wiki Updates](https://wiki.servarr.com)

### CODE REVIEW

File: src/NzbDrone.Core/Indexers/Exceptions/IndexerException.cs [hunk 1]
```diff
@@ -18,6 +18,16 @@ public IndexerException(IndexerResponse response, string message)
             _indexerResponse = response;
         }
 
+        public IndexerException(string message)
+            : base(message)
+        {
+        }
```
Comment (Qstick) [inline src/NzbDrone.Core/Indexers/Exceptions/IndexerException.cs:24]: Allowing initialization without setting response may result in unwanted behavior downstream in history processing. HttIndexerBase query error handling expects a response to be present when handling IndexerExceptions

Comment (bakerboy448) [reply src/NzbDrone.Core/Indexers/Exceptions/IndexerException.cs:25]: Added

Comment (bakerboy448) [reply src/NzbDrone.Core/Indexers/Exceptions/IndexerException.cs:25]: hmm actually needed to add that as the existing

```chsarp
        public IndexerException(IndexerResponse response, string message)
            : base(message)
        {
            _indexerResponse = response;
        }
```
was causing VS to not like it

thoughts?

Comment (bakerboy448) [reply src/NzbDrone.Core/Indexers/Exceptions/IndexerException.cs:25]: builds DOA pending figuring out how to handle these; 

we need to not require the response to be ingested or change the cardigann exceptions to include the responses

**Comment by JigSawFr** [comment]
Nice addition, will be useful!

**Comment by bakerboy448** [comment]
covered elsehwhere
https://github.com/Prowlarr/Prowlarr/pull/825

**Pull request closed without merging**